lib/net.js

```diff
@@ -894,6 +906,12 @@ Socket.prototype.connect = function(options, cb) {
   self._connecting = true;
   self.writable = true;
 
+  // Cork then uncork upon connecting.
+  this.cork();
+  this.once('connect', function() {
+    this.uncork();
+  });
```

Comment:
hm. wonder how this would have worked before. I assume writes to an unconnected socket would have either errored or been lost.